### PR TITLE
test/e2e_kubeadm: fix matching UnversionedKubeletConfigMap defaults

### PR DESCRIPTION
**Pull request by neolit123**
/kind failing-test

#### What this PR does / why we need it:

The kubeadm defaults in features.go differ between versions.
e2e_kubeadm tests cannot import the kubeadm features.go,
or easily detect the versioned of the kubeadm binary used
to create the cluster.

Check for the existence of both versioned and unversioned
objects independent of the value of the FG. Once the FG
goes GA only the unversioned objects should be checked.

Without this change kubeadm e2e skew tests will fail where kubeadm
is at 1.24 (has the FG defaulted to true), the FG is not
explicitly set by the user and the k8s version is at 1.23.

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubeadm/issues/1582#issuecomment-1040286673
https://k8s-testgrid.appspot.com/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-latest-on-1-23

#### Special notes for your reviewer:
need to backport this to 1.23. technically we only need to fix the 1.23 branch, but having this in master first is accurate.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_kubeadm/kubelet_config_test.go [hunk 1]
```diff
@@ -37,12 +38,12 @@ var (
 	kubeletConfigRoleName        string
 	kubeletConfigRoleBindingName string
 
-	kubeletConfigConfigMapResource = &authv1.ResourceAttributes{
-		Namespace: kubeSystemNamespace,
-		Name:      "",
-		Resource:  "configmaps",
-		Verb:      "get",
-	}
```
Comment (neolit123) [inline test/e2e_kubeadm/kubelet_config_test.go:45]: this object was unused

File: test/e2e_kubeadm/kubelet_config_test.go [hunk 2]
```diff
@@ -79,46 +80,54 @@ var _ = Describe("kubelet-config ConfigMap", func() {
...
-
+		// TODO: switch to GetConfigMap once UnversionedKubeletConfigMap feature gate goes GA:
+		// https://github.com/kubernetes/kubeadm/issues/1582
+		cm, err := f.ClientSet.CoreV1().
+			ConfigMaps(kubeSystemNamespace).
+			Get(context.TODO(), kubeletConfigConfigMapName, metav1.GetOptions{})
+		if err != nil {
+			cm, err = f.ClientSet.CoreV1().
+				ConfigMaps(kubeSystemNamespace).
+				Get(context.TODO(), kubeletConfigConfigMapNameVersioned, metav1.GetOptions{})
```
Comment (pacoxu) [inline test/e2e_kubeadm/kubelet_config_test.go:101]: just remove lines like this after the feature gate is removed.

/lgtm
hadn't thought about the kubeadm n-1 kubelet skew when reviewing last PR.
/unhold

**Pull request merged**